rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMService.java

```diff
@@ -151,6 +151,23 @@ String editNodeSource(String sessionId, String nodeSourceName, String infrastruc
             String[] policyParameters, String[] policyFileParameters, String nodesRecoverable)
             throws RestServerException, ServiceException;
 
+    /**
+     * Override the dynamic parameters of a NodeSource
+     * @param sessionId current session
+     * @param nodeSourceName name of the NS to edit
+     * @param infrastructureType infrastructure manager full class name
+     * @param infrastructureParameters IM String parameters, null value for files
+     * @param infrastructureFileParameters file parameters
+     * @param policyType policy full class name
+     * @param policyParameters String parameters, null value for files
+     * @param policyFileParameters file parameters
+     * @throws RestServerException
+     * @throws ServiceException
+     */
+    String updateDynamicParameters(String sessionId, String nodeSourceName, String infrastructureType,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)